core/src/test/java/io/questdb/test/client/LineSenderBuilderTest.java

```diff
@@ -150,6 +188,61 @@ public void testBufferSizeDoubleSet() throws Exception {
+            assertConfStrError("http::addr=localhost:8080;auto_flush_rows=notanumber;", "invalid auto_flush_rows [value=notanumber]");
+            assertConfStrError("http::addr=localhost:8080;auto_flush=invalid;", "invalid auto_flush [value=invalid, allowed-values=[on, off]]");
+            assertConfStrError("http::addr=localhost:8080;auto_flush=off;auto_flush_rows=100;", "cannot set auto flush rows when auto-flush is disabled");
+            assertConfStrError("http::addr=localhost:8080;auto_flush_rows=100;auto_flush=off;", "auto flush rows was already configured [autoFlushRows=100]");
+            assertConfStrError(Sender.builder().http(), "http::addr=localhost:8080;auto_flush_rows=100;auto_flush=off;", "protocol was already configured [protocol=http]");
+            assertConfStrError(Sender.builder().tcp(), "http::addr=localhost:8080;auto_flush_rows=100;auto_flush=off;", "protocol was already configured [protocol=tcp]");
+            assertConfStrError(Sender.builder().address("remote"), "http::addr=localhost:8080;auto_flush_rows=100;auto_flush=off;", "server address was already configured [address=remote]");
+            assertConfStrError(Sender.builder().address("remote:1234"), "http::addr=localhost:8080;auto_flush_rows=100;auto_flush=off;", "server address was already configured [address=remote]");
+            assertConfStrError(Sender.builder().port(1234), "http::addr=localhost:8080;auto_flush_rows=100;auto_flush=off;", "server port was already configured [port=1234]");
+            assertConfStrError(Sender.builder().advancedTls().disableCertificateValidation(), "http::addr=localhost:8080;tls_verify=on;", "tls_verify was already configured");
+            assertConfStrError(Sender.builder().enableTls(), "http::addr=localhost:8080;", "cannot use http protocol when TLS is enabled. use https instead");
+            assertConfStrError(Sender.builder().enableTls(), "tcp::addr=localhost:8080;", "cannot use tcp protocol when TLS is enabled. use tcps instead");
+
+            assertConfStrOk(Sender.builder().enableTls(), "https::addr=localhost:8080;");
+            assertConfStrOk("http::addr=localhost:8080;auto_flush=on;auto_flush_rows=100;");
+            assertConfStrOk("http::addr=localhost:8080;auto_flush_rows=100;auto_flush=on;");
+            assertConfStrOk("http::addr=localhost;auto_flush=on;");
+            assertConfStrOk("http::addr=localhost;auto_flush=off;");
+            assertConfStrOk("http::addr=localhost;");
+            assertConfStrOk("http::addr=localhost:8080;");
+            assertConfStrOk("http::addr=localhost:8080;token=foo;");
+            assertConfStrOk("http::addr=localhost:8080;token=foo;retry_timeout=1000;max_buf_size=1000000;");
+            assertConfStrOk("https::addr=localhost:8080;tls_verify=unsafe_off;auto_flush_rows=100;");
+            assertConfStrOk("https::addr=localhost:8080;tls_verify=on;");
```

Comment:
nit: would be good to test with IPv6 address and check the error is adequate

Reply:
nit: would be good to test with token strings which include `==` (double equals) in the middle, end of config string.
Generally, I see the tests specify config string in some order, addr first then options, is the random order supported?

Reply:
random order is supported. let me add some tests for this too.